src/ooxml/java/org/apache/poi/POIXMLDocumentPart.java

```diff
@@ -615,49 +640,55 @@ protected void read(POIXMLFactory factory, Map<PackagePart, POIXMLDocumentPart>
         if (otherChild != null && otherChild != this) {
             throw new POIXMLException("Unique PackagePart-POIXMLDocumentPart relation broken!");
         }
-
+        
         if (!pp.hasRelationships()) return;
-
+        
         PackageRelationshipCollection rels = packagePart.getRelationships();
         List<POIXMLDocumentPart> readLater = new ArrayList<>();
-
+        
         // scan breadth-first, so parent-relations are hopefully the shallowest element
         for (PackageRelationship rel : rels) {
             if(rel.getTargetMode() == TargetMode.INTERNAL){
                 URI uri = rel.getTargetURI();
-
+                
                 // check for internal references (e.g. '#Sheet1!A1')
                 PackagePartName relName;
                 if(uri.getRawFragment() != null) {
                     relName = PackagingURIHelper.createPartName(uri.getPath());
                 } else {
                     relName = PackagingURIHelper.createPartName(uri);
                 }
-
+                
                 final PackagePart p = packagePart.getPackage().getPart(relName);
                 if (p == null) {
                     logger.log(POILogger.ERROR, "Skipped invalid entry " + rel.getTargetURI());
                     continue;
                 }
-
+                
                 POIXMLDocumentPart childPart = context.get(p);
                 if (childPart == null) {
                     childPart = factory.createDocumentPart(this, p);
+                    //here we are checking if part if embedded and excel then set it to chart class
+                    //so that at the time to writing we can also write updated embedded part
+                    if(this instanceof XDDFChart && childPart instanceof XSSFWorkbook)
```

Comment:
This is a code smell...

After setting the `parent` on the `childPart`, some method should be called on the `childPart` which would then call the `setWorkbook` method.